python/mxnet/autograd.py

```diff
@@ -0,0 +1,68 @@
+# coding: utf-8
+"""Autograd for NDArray."""
+from __future__ import absolute_import
+from __future__ import division
+
+import ctypes
+import functools
+from .base import _LIB, check_call
+from .base import mx_uint, NDArrayHandle, c_array
+from .ndarray import NDArray
+from .ndarray import array as ndarray
+
+def set_recording(recording):
+    check_call(_LIB.MXAutogradSetRecording(
+        ctypes.c_int(recording)))
+
+def set_mark_for_record(arrays, mark):
+    handles = []
+    for arr in arrays:
+        handles.append(arr.handle)
+    check_call(_LIB.MXAutogradSetMarkForRecord(
+        len(handles),
+        c_array(NDArrayHandle, handles),
+        ctypes.c_int(mark)))
+
+def compute_gradient(inputs, outputs):
+    input_handles = []
```

Comment:
Inputs here is all inputs used for computing forward, but you may only want gradients of part of them. Consider:
```python

f_partial(a, b, c, d):
  e0 = a * b
  e1 = b * c
  e2 = e0 * e1
  e3 = e2 * d

grad_func = grad(f_partial, [3])
grad_vals = grad_func(a, b, c, d)
```

You only want the gradient of `d` here, but you have to give `a, b, c` in order to compute the gradient.